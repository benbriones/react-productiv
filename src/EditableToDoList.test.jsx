import { describe, it, expect } from "vitest";

import React from "react";
import { fireEvent, queryByText, render } from "@testing-library/react";
import EditableTodoList from "./EditableTodoList";


describe("EditableToDoList", function () {
    const testTodos =
        [{ id: 4, title: "celebrate bday", description: "for dinner", priority: 1 },
        { id: 5, title: "eat pocky", description: "mens pocky only", priority: 2 },
        { id: 6, title: "get dessert", description: "afters only", priority: 3 }];

    it("renders without crashing", function () {

        render(<EditableTodoList todos={testTodos} />);

    });

    it("matches snapshot", function () {

        const { container } = render(<EditableTodoList todos={testTodos} />);
        expect(container).toMatchSnapshot();

    });



    it ("contains all editable todos", function () {
        // const { container } = render(<EditableTodoList todos={testTodos} />)
        const rendered = render(<EditableTodoList todos={testTodos} />)

        expect(rendered.queryByText("celebrate bday")).toBeInTheDocument
        expect(rendered.queryByText("eat pocky")).toBeInTheDocument
        expect(rendered.queryByText("get dessert")).toBeInTheDocument


        const editBtns = rendered.queryAllByText('Edit')
        expect(editBtns.length).toEqual(3)
        // expect()


      });





});