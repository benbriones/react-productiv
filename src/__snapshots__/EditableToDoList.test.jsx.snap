// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`EditableToDoList > matches snapshot 1`] = `
<div>
  <div
    class="EditableTodoList"
  >
    <div
      class="EditableTodo"
    >
      <div
        class="mb-3"
      >
        <div
          class="float-end text-sm-end"
        >
          <button
            class="EditableTodo-toggle btn-link btn btn-sm"
          >
            Edit
          </button>
          <button
            class="EditableTodo-delBtn btn-link btn btn-sm text-danger"
          >
            Del
          </button>
        </div>
        <div
          class="Todo"
          id="4"
        >
          <div>
            <b>
              celebrate bday
            </b>
             
            <small>
              (Priority: 
              1
              )
            </small>
          </div>
          <div>
            <small>
              for dinner
            </small>
          </div>
        </div>
      </div>
    </div>
    <div
      class="EditableTodo"
    >
      <div
        class="mb-3"
      >
        <div
          class="float-end text-sm-end"
        >
          <button
            class="EditableTodo-toggle btn-link btn btn-sm"
          >
            Edit
          </button>
          <button
            class="EditableTodo-delBtn btn-link btn btn-sm text-danger"
          >
            Del
          </button>
        </div>
        <div
          class="Todo"
          id="5"
        >
          <div>
            <b>
              eat pocky
            </b>
             
            <small>
              (Priority: 
              2
              )
            </small>
          </div>
          <div>
            <small>
              mens pocky only
            </small>
          </div>
        </div>
      </div>
    </div>
    <div
      class="EditableTodo"
    >
      <div
        class="mb-3"
      >
        <div
          class="float-end text-sm-end"
        >
          <button
            class="EditableTodo-toggle btn-link btn btn-sm"
          >
            Edit
          </button>
          <button
            class="EditableTodo-delBtn btn-link btn btn-sm text-danger"
          >
            Del
          </button>
        </div>
        <div
          class="Todo"
          id="6"
        >
          <div>
            <b>
              get dessert
            </b>
             
            <small>
              (Priority: 
              3
              )
            </small>
          </div>
          <div>
            <small>
              afters only
            </small>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
